cli/src/commands/ledger.rs

```diff
@@ -2003,6 +2054,172 @@ pub fn get_active_transaction_author_agreement(_pool_handle: i32) -> Result<Opti
+                .map_err(|err| println_err!("Invalid transaction response: {:?}", err))
+        )
+}
+
+pub fn set_request_fees(ctx: &CommandContext,
+                        request: &mut String,
+                        wallet_handle: i32,
+                        submitter_did: Option<&str>,
+                        source_payment_address: Option<&str>,
+                        fee: Option<u64>,
+                        fees_inputs: Option<Vec<&str>>,
+                        fees_outputs: Option<Vec<String>>,
+                        extra: Option<&str>) -> Result<Option<String>, ()> {
+    if source_payment_address.is_none() && fees_inputs.is_none() {
+        return Ok(None);
+    }
+
+    if source_payment_address.is_some() && fees_inputs.is_some() {
+        return Err(println_err!("Only one of `source_payment_address`, `fees_inputs` can be specified."));
+    }
+
+    let (inputs, outputs) = match (source_payment_address, fee) {
+        (Some(source_), Some(fee_)) => {
+            build_payment_sources_for_addresses(ctx, source_, None, None, Some(fee_))?
+        }
+        (Some(_), None) => { return Err(println_err!("Fee value must be specified together with `source_payment_address`.")); }
+        _ => {
+            match fees_inputs {
+                Some(inputs_) => {
+                    let inputs = inputs_.into_iter().map(String::from).collect();
+                    let outputs =
+                        fees_outputs
+                            .as_ref()
+                            .ok_or(())
+                            .and_then(|outputs_| parse_payment_outputs(outputs_))
+                            .unwrap_or_default();
+                    (inputs, outputs)
+                }
+                _ => { return Ok(None); }
+            }
+        }
+    };
+
+    let inputs_json = serialize(&inputs)?;
+    let outputs_json = serialize(&outputs)?;
+
+    let (req_with_fees, payment_method) = Payment::add_request_fees(wallet_handle, submitter_did, request, &inputs_json, &outputs_json, extra)
+        .map_err(|err| handle_payment_error(err, None))?;
+
+    *request = req_with_fees;
+
+    Ok(Some(payment_method))
+}
+
+fn prepare_sources_for_payment_cmd(ctx: &CommandContext,
+                                   source_payment_address: Option<String>,
+                                   target_payment_address: Option<String>,
+                                   amount: Option<u64>,
+                                   fee: Option<u64>,
+                                   inputs: Option<Vec<&str>>,
+                                   outputs: Option<Vec<String>>) -> Result<(String, String), ()> {
+    let (inputs, outputs) = match (source_payment_address, target_payment_address, amount) {
+        (Some(source_address), Some(target_address), Some(amount_)) => {
+            build_payment_sources_for_addresses(&ctx, &source_address, Some(&target_address), Some(amount_), fee)?
+        }
+        _ => {
```

Comment:
I suggest to split this `_` to 2 path: None None None => { current logic}, _ => error